gantry/firmware/step_motor.cpp

```diff
@@ -11,6 +11,11 @@ static auto handler_class = motor_handler::MotorInterruptHandler<
 
 void step_motor() {
     if (handler_class.pulse()) {
+        if (handler_class.set_direction_pin(get_direction_bit())) {
```

Comment:
wait hold on. we can just make this fully dependent on the velocity. we don't have to read the gpio's current status at all. all we have to do is `if (current_move.velocity > 0 ) { turn_on_direction_pin(); } else { turn_off_direction_pin(); }`. it doesn't matter if we set it every tick, it's just writing a bit. lets us keep it super super simple